common/djangoapps/third_party_auth/models.py

```diff
@@ -161,7 +161,11 @@ def clean(self):
     def provider_id(self):
         """ Unique string key identifying this provider. Must be URL and css class friendly. """
         assert self.prefix is not None
-        return "-".join((self.prefix, ) + tuple(getattr(self, field) for field in self.KEY_FIELDS))
+        fields = []
```

Comment:
Thankfully, this change is no longer necessary as I took a different approach. Modifying provider_id was a bit too risky anyway.